src/zmalloc.c

```diff
@@ -436,9 +460,18 @@ char *zstrdup(const char *s) {
 }
 
 size_t zmalloc_used_memory(void) {
-    size_t um;
-    atomicGet(used_memory,um);
-    return um;
+    size_t local_num_active_threads;
+    size_t totol_mem = 0;
+    atomicGet(num_active_threads,local_num_active_threads);
+    if (local_num_active_threads > MAX_THREADS) {
+        local_num_active_threads = MAX_THREADS;
+    }
+    for (size_t i = 0; i < local_num_active_threads; ++i) {
+        size_t thread_used_mem;
```

Comment:
Should this one be `long long` as well? If I understand correctly, there is a chance that `used_memory[i].used_memory` might be negative, if so, reading negative into size_t will give us wrong result.

Reply:
@tezc haha, the result will eventually be revised.
like
```
size_t n = -1;
n += 2
now n is 1.
```

Reply:
@tezc you are right, if it is in 32 bits it will be problematic, the value will be truncated.

Reply:
Isn't this also possible: 

```
size_t n = -1000;
n += 2

// Get used memory at this point
zmalloc_used_memory(); // returns a huge number
...
```

Reply:
no, if we treat the whole use_memory array as a whole, it doesn't end up with a negative.

Reply:
Okay, you're right. 
I can't wrap my head around what the result would be if there is bunch of overflows in the addition of several numbers :) 

@MeirShpilraien Probably, `totol_mem` should be `long long` as well. Otherwise, we'll have same problem there (if there is a problem)